selection_sort.go

```diff
@@ -0,0 +1,16 @@
+package sort
+
+func selectionSort(arr []int) []int {
```

Comment:
### Prerequisite Checks

Ensure that the array `arr` is not nil before entering the sorting function.

### Runtime Errors

No specific runtime errors are present in this line.

### Optimizations

No specific optimizations are needed for this line.

### Security Issues

No specific security issues are present in this line.